### PR TITLE
add string_view support

### PR DESCRIPTION
**Pull request by SlaaneshChosen**

### CODE REVIEW

**Comment by SlaaneshChosen** [comment]
`string` and `string_view` are two different things when parsing a json. For a `string`, you copy it from the orginal json string and handle escaped. Sometimes you may just want to have a `view` of a field and will never change it, and `string_view` is a better choince.

File: include/glaze/json/read.hpp [hunk 1]
```diff
@@ -394,6 +394,92 @@ namespace glz
...
+               switch (*it) {
+               case '"':
+               case '\\':
+               case '/':
+               case 'b':
+               case 'f':
+               case 'n':
+               case 'r':
+               case 't':
+               case 'u': {
```
Comment (mwalcott3) [inline include/glaze/json/read.hpp:426]: Should probably validate \u is followed by 4 hex digits when force_conformance == true. But I can make that change later.

File: tests/json_test/json_test.cpp [hunk 2]
```diff
@@ -64,6 +64,35 @@ suite starter = [] {
    };
 };
 
+//struct my_strings
```
Comment (mwalcott3) [inline tests/json_test/json_test.cpp]: Why is this test commented out?

**Review by mwalcott3 — APPROVED** [review]
Looks good. I should be talking to @stephenberry soon so Ill probably wait for his input before we merge this.

**Pull request merged**